### PR TITLE
Extend UNNEST documentation with usage examples

### LINKED ISSUE #3812: Clarify unnest and cross join behavior
Copied from slack conversation with a user, @martint and @findepi : 

Hello community,
I'm running a query with a select on a field "addresses" that is a nested array (we use the Elasticsearch connector) that returns all the records in the database, the records that have a value in the field and those that have a null value.
    select addresses from customer_contacts;
But, if I run this query but with a CROSS JOIN UNNEST to the same field "addresses", only the registers with a non null value in the field are returned.
    select addresses from customer_contacts cross join unnest(addresses);
This is problem with Presto? By the docs, the CROSS JOIN should return all combinations of the values, which, in my understanding, includes null values...

martin  6 hours ago
If the addresses field is empty or null, UNNEST produces 0 rows, so the CROSS JOIN would produce 0 rows for that entry.
What you want is:
SELECT addresses FROM customer_contacts LEFT JOIN UNNEST(addresses) ON true

martin  6 hours ago
cc @Manfred Moser, we may want to add something to the docs explaining this scenario

findepi  5 hours ago
People are so used that UNNEST “equals” to CROSS JOIN UNNEST (because it used to be so).
This is why we should highlight the LEFT JOIN UNNEST more than we otherwise would need to. This in fast should be our leading examples, and perhaps default “unnest paradigm”.

martin  4 hours ago
Indeed. UNNEST  is just a table function, and as any other table-producing entity, it can be used in the FROM clause or as part of a JOIN clause. The regular semantics of  JOIN apply.

### LINKED ISSUE #8471: UNNEST left join docs
https://trino.io/docs/current/sql/select.html?highlight=unnest#unnest  UNNEST needs an example with 'LEFT JOIN ON TRUE' instead of 'CROSS JOIN' so rows don't get dropped

### PR DESCRIPTION
**Pull request by findinpath**
This PR addresses:

fixes #3812
fixes #8471

documentation  issues

### CODE REVIEW

**Review by kasiafi — APPROVED** [review]
LGTM. @mosabua could you please take a look?

File: docs/src/main/sphinx/sql/select.rst [hunk 1]
```diff
@@ -1027,21 +1027,123 @@ UNNEST
...
+.. code-block:: text
+
+      a   | b | rownumber
+    ------+---+-----------
+        2 | 7 |         1
+        5 | 8 |         2
+     NULL | 9 |         3
+    (3 rows)
+
+``UNNEST`` will return 0 entries when the array/map to be expanded is empty::
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/select.rst:1102]: returns zero entries when

Comment (mosabua) [reply docs/src/main/sphinx/sql/select.rst:1102]: .. could also leave "to be expanded" out altogether

File: docs/src/main/sphinx/sql/select.rst [hunk 2]
```diff
@@ -1027,21 +1027,123 @@ UNNEST
...
+
+    SELECT * FROM UNNEST (ARRAY[]) AS t(value);
+
+.. code-block:: text
+
+     value
+    -------
+    (0 rows)
+
+``UNNEST`` will return 0 entries when the array/map to be expanded is null::
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/select.rst:1112]: returns zero entries when

File: docs/src/main/sphinx/sql/select.rst [hunk 3]
```diff
@@ -1027,21 +1027,123 @@ UNNEST
...
+    SELECT * FROM UNNEST (CAST(null AS ARRAY(integer))) AS t(number);
+
+.. code-block:: text
 
-Using a single column::
+     number
+    --------
+    (0 rows)
+
+``UNNEST`` is normally used with a ``JOIN`` and can reference columns
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/select.rst:1122]:
```
``JOIN``, and can 
```

**Review by mosabua — CHANGES_REQUESTED** [review]
Minor tweaks only .. then good to go imho

**Comment by findinpath** [comment]
@mosabua  can you please have another look on this one?

**Review by mosabua — APPROVED** [review]
Content is great. Ideally you could wrap the paragraphs at 80 column widths... but thats source code cosmetics..

**Pull request merged**